app/main/posts/views/post-view-data.directive.js

```diff
@@ -88,6 +88,36 @@ function PostViewDataController(
      */
     var newPostsAfter = moment().utc();
     $scope.savingPost = {saving: false};
+
+    $rootScope.$on('event:edit:post:data:mode:saveSuccess', function () {
+        function removePostFromList() {
+            for (var i = 0; i < $scope.posts.length; i++) {
```

Comment:
I'd generally suggest just using underscore for this ie. 
```
let index = _.findIndex($scope.posts, { id: selectedId });
$scope.posts.splice(i, 1);
```
Unless we see a real slowdown from doing this, I favour readability